Pricenode: Add JVM flag ExitOnOutOfMemoryError

alkum opened the pull request:
Add flag that causes the JVM to exit with exit code 3 whenever there is an OutOfMemory exception for the pricenode service, which causes the systemd bisq-pricenode service to restart.

Can be tested by running the pricenode on a VM with very little memory, or by artificially limiting the JVM memory in addition to this flag: `-Xmx10m -XX:+ExitOnOutOfMemoryError`. The tests should result in an endless restart cycle for the pricenode service (`sudo systemctl status bisq-pricenode` should show service uptime of max 1-2 seconds each time it's called).

Fixes #6225

alkum committed:
Add JVM flag ExitOnOutOfMemoryError

Add flag that causes the JVM to exit with exit code 3 whenever there is an OutOfMemory exception.

ghost commented:
You could do the same for seednode, to resolve issue #6225.

https://github.com/bisq-network/bisq/blob/ad15fcd5c4947325dfbcca16c7ec940cc85ec8b3/seednode/bisq.env#L1-L8

alkum committed:
Add ExitOnOutOfMemoryError JVM flag tp seednode

Flag ensures the seednode exists with error code 3 when an OutOfMemory exception happens.

alkum commented:
Done.

ripcurlx reviewed (approved):
utACK

wiz commented:
Thank you 🙏
